dao/src/main/java/org/thingsboard/server/dao/sql/query/DefaultQueryLogComponent.java

```diff
@@ -33,8 +40,79 @@
     @Override
     public void logQuery(QueryContext ctx, String query, long duration) {
         if (logSqlQueries && duration > logQueriesThreshold) {
+
+            String sqlToUse = substituteParametersInSqlString(query, ctx);
+
             log.info("QUERY: {} took {} ms", query, duration);
+            log.info("QUERY SQL TO USE: {} took {} ms", sqlToUse, duration);
+
             Arrays.asList(ctx.getParameterNames()).forEach(param -> log.info("QUERY PARAM: {} -> {}", param, ctx.getValue(param)));
         }
     }
+
+    String substituteParametersInSqlString(String sql, SqlParameterSource paramSource) {
+
+        ParsedSql parsedSql = NamedParameterUtils.parseSqlStatement(sql);
+        List<SqlParameter> declaredParams = NamedParameterUtils.buildSqlParameterList(parsedSql, paramSource);
+        StringBuilder actualSql = new StringBuilder(parsedSql.toString());
+
+        if (declaredParams.isEmpty()) {
+            return sql;
+        }
+
+        for (SqlParameter parSQL: declaredParams) {
+            String paramName = parSQL.getName();
+            if (!paramSource.hasValue(paramName)) {
+                continue;
+            }
+
+            Object value = paramSource.getValue(paramName);
+            if (value instanceof SqlParameterValue) {
+                value = ((SqlParameterValue)value).getValue();
+            }
+
+            if (!(value instanceof Iterable)) {
+
+                String ValueForSQLQuery = getValueForSQLQuery(value);
+                String sqlTemp = sql.replace(":" + paramName, ValueForSQLQuery);
+                sql = sqlTemp;
+                continue;
+            }
+
+            //Iterable
+            int count = 0;
+            String valueArrayStr = "";
+
+            for (Object valueTemp: (Iterable)value) {
+
+                if (count > 0) {
+                    valueArrayStr+=", ";
+                }
+
+                String ValueForSQLQuery = getValueForSQLQuery(valueTemp);
+                valueArrayStr += ValueForSQLQuery;
+                ++count;
+            }
+
+            if (!valueArrayStr.isEmpty()){
+                String sqlTemp = sql.replace(":" + paramName, valueArrayStr);
+                sql = sqlTemp;
+            }
+        }
+
+        return sql;
+    }
+
+    String getValueForSQLQuery(Object valueParameter) {
+
+        if (valueParameter instanceof String) {
+            return "'" + valueParameter + "'";
```

Comment:
The string must be SQL safe. For example, user can search the character `'` or `\n` or CRLF
In that case, SQL syntax will be violated. Probably, there are more use cases. Mostly wanted approach is to add more test cases that may break the SQL query after substitution.